### PR TITLE
Add request library

### PR DESCRIPTION
**Pull request by Ashvin-Ranjan**

### CODE REVIEW

File: python/libraries/request.py [hunk 1]
```diff
@@ -28,4 +28,4 @@ def post(args):
 
 
 def _values():
-	return {"post": {"code": "int", "reason": "str", "text": "str"}, "get": {"code": "int", "reason": "str", "text": "str"}}
+	return {"post": ("str", {"code": "int", "reason": "str", "text": "str"}), "get": ("str", {"code": "int", "reason": "str", "text": "str"})}
```
Comment (SheepTester) [inline python/libraries/request.py:31]: epic, though ideally they'd be async and return `cmd`s (you can mimic discordn.py on the main branch), but I guess I can fix that later

File: python/scope.py [hunk 2]
```diff
@@ -767,7 +767,7 @@ def type_check_expr(self, expr):
 			func_type = self.type_check_expr(function)
 			if func_type is None:
 				return None
-			if not isinstance(func_type, tuple):
+			if not isinstance(func_type, tuple) and not isinstance(func_type, NModule):
```
Comment (SheepTester) [inline python/scope.py:770]: ??? Why would you allow calling modules? If I'm not mistaken, NModules are (they extend) dictionaries

Comment (Ashvin-Ranjan) [reply python/scope.py:770]: That was a mistake, ill remove it in the next commit

File: python/libraries/request.py (hunk 1 above)
Comment (Ashvin-Ranjan) [reply python/libraries/request.py:31]: Ok ill do that in a bit, but before that we need hashtables or hashmaps, whatever they called

**Review by SheepTester — APPROVED** [review]

**Pull request merged**